gnome-shell.d.ts

```diff
@@ -0,0 +1,334 @@
+/// <reference path="./node_modules/@gtile/gjs/dom.d.ts" />
```

Comment:
It's essentially the equivalent of a custom library that would otherwise be included in the [`compilerOptions.lib`](https://www.typescriptlang.org/tsconfig#lib) field of `tsconfig.json`. We are not in a browser environment, so I can't use the default TypeScript `DOM` type definitions. But since we rely on some functions in the global namespace (namely `setTimeout` and `clearTimeout`) we to reference the global namespace type declarations of GJS.

Personally, I wouldn't add a dedicated comment for that. I think the `declare global` section inside `dom.d.ts` speaks for itself.